client/src/dataset/components/ScanSize.tsx

```diff
@@ -0,0 +1,70 @@
+import { Field } from "formik";
+import * as React from "react";
+import { Button, Form, FormFieldProps } from "semantic-ui-react";
+
+interface ScanSizeProps {
+    value: string,
+    minScan: number,
+    maxScan: number,
+    setFieldValue: (field: string, value: any, shouldValidate?: boolean) => void,
+}
+
+const ScanSize: React.FC<ScanSizeProps> = ({ value, minScan, maxScan, setFieldValue }) => {
+
+    const [scanSize, setScanSize] = React.useState(value.split(",").length>=minScan? value.split(","): new Array(minScan).fill(""));
+
+    const scanRefsArray = React.useRef<FormFieldProps[]>([]);
+
+    const onScanSizeChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+      const idx = parseInt(e.target.name.split("_")[2], 10);
+      scanSize[idx] = e.target.value.toString();
+      setScanSize(scanSize);
+      setFieldValue("scan_size", scanSize.toString());
+    };
+
+    const onCommaPress = (e: KeyboardEvent) => {
+        const idx = parseInt((e.target as HTMLInputElement).name.split("_")[2], 10);
+        if(e.keyCode === 188){
+            idx===(scanSize.length-1)? newScanDim() : scanRefsArray.current[idx+1].focus();
```

Comment:
This is really nice behavior :+1: Note that for conditions, it is always better to use a proper `if` statement, if possible. Inside JSX expressions, it is often not possible, so using the tenary operator is fine, but here, a proper `if` is much better readable.